tests/unit/test_generate.py

```diff
@@ -0,0 +1,215 @@
+import json
+import os
+import subprocess
+
+import flexmock
+import pytest
+
+
+from apigentools.commands.generate import GenerateCommand
+from apigentools.config import Config, LanguageConfig
+from apigentools.utils import run_command
+
+FIXTURE_DIR = os.path.join(os.path.dirname(os.path.realpath(__file__)), "fixtures")
+
+
+def test_get_missing_templates(tmpdir):
+    temp_dir = tmpdir.mkdir("missing_templates")
+    languages = ["python", "go", "ruby"]
+    args = flexmock(template_dir=temp_dir)
+    cmd = GenerateCommand({}, args)
+    missing = cmd.get_missing_templates(languages)
+    assert missing == languages
+
+
+def test_get_version_from_lang_oapi_config():
+    args = flexmock()
+    cmd = GenerateCommand({}, args)
+    with open(os.path.join(FIXTURE_DIR, "go_oapi_config.json")) as f:
+        oapi_config = json.loads(f.read())
+    result = cmd.get_version_from_lang_oapi_config(oapi_config)
+    assert result == "0.0.1"
+
+
+def test_get_image_name():
+    args = flexmock(generated_with_image="apigentools:latest")
+    cmd = GenerateCommand({}, args)
+    result = cmd.get_image_name()
+    assert result == "apigentools:latest"
+
+
+def test_get_codegen_version():
+    args = flexmock()
+    with open(os.path.join(FIXTURE_DIR, "raw_dict.json"), "r") as f:
+        raw_dict = json.loads(f.read())
+    cfg = Config(raw_dict)
+    cmd = GenerateCommand(cfg, args)
+    expected_result = run_command([cmd.config.codegen_exec, "version"])
+    expected_result = expected_result.stdout.strip()
+    # this is the version of openapi-generator we are using
+    assert cmd.get_codegen_version() == expected_result
+
+
+def test_get_stamp():
+    args = flexmock(generated_with_image=None, spec_repo_dir=".", additional_stamp=[])
+    with open(os.path.join(FIXTURE_DIR, "raw_dict.json"), "r") as f:
+        raw_dict = json.loads(f.read())
+    cfg = Config(raw_dict)
+    cmd = GenerateCommand(cfg, args)
+    stamp = cmd.get_stamp()
+    assert "Generated with" in stamp
+    assert "codegen version" in stamp
```

Comment:
Can we use the codegen version the same way you did above (like `run_command([cmd.config.codegen_exec, "version"])` )

Reply:
I added a way to use the codegen version, let me know if it's what you had in mind